android/src/main/java/com/magicleap/magicscript/ar/ModelRenderableLoaderImpl.kt

```diff
@@ -20,22 +20,43 @@ import android.content.Context
 import android.net.Uri
 import com.google.ar.sceneform.assets.RenderableSource
 import com.google.ar.sceneform.rendering.ModelRenderable
+import com.magicleap.magicscript.utils.Utils
 import com.magicleap.magicscript.utils.logMessage
 
 class ModelRenderableLoaderImpl(private val context: Context) : ModelRenderableLoader {
 
-    override fun loadRenderable(modelUri: Uri, resultCallback: (result: RenderableResult) -> Unit) {
-        ModelRenderable.builder()
-            .setSource(
-                context, RenderableSource.builder().setSource(
-                    context,
-                    modelUri,
-                    RenderableSource.SourceType.GLB
-                ) // GLB (binary) or GLTF (text)
+    override fun loadRenderable(
+        modelUri: Uri,
+        resultCallback: (
+            result: RenderableResult<ModelRenderable>
+        ) -> Unit
+    ) {
+        val modelType = Utils.detectModelType(modelUri, context)
+        if (modelType == ModelType.UNKNOWN) {
+            val errorMessage = "Unresolved model type"
+            logMessage(errorMessage, true)
+            resultCallback(RenderableResult.Error(Exception(errorMessage)))
+            return
+        }
+
+        val builder = ModelRenderable.builder()
+
+        if (modelType == ModelType.GLB) {
+            builder.setSource(
+                context, RenderableSource.builder()
+                    .setSource(
+                        context,
+                        modelUri,
+                        RenderableSource.SourceType.GLB // GLB (binary) or GLTF (text)
+                    )
                     .setRecenterMode(RenderableSource.RecenterMode.CENTER)
                     .build()
```

Comment:
here is `.build` and in line 60 as well, is it correct?

Reply:
Yes, it seems incorrect, but the first build is related to `RenderableSource.builder()` but the second  is for `ModelRenderable.builder()`